core/src/main/scala/org/apache/spark/deploy/rest/RestClient.scala

```diff
@@ -219,14 +221,20 @@ private[deploy] class StandaloneRestClient extends Logging {
 
   /** Return the base URL for communicating with the server, including the protocol version. */
   private def getBaseUrl(master: String): String = {
-    val masterUrl = master.stripPrefix("spark://").stripSuffix("/")
+    val masterUrl = if (master.startsWith("spark://")) {
+      master.stripPrefix("spark://").stripSuffix("/")
+    } else {
+      master.stripPrefix("mesos://").stripSuffix("/")
+    }
```

Comment:
(update: as discussed offline we'll just remove the check in `SparkSubmitArguments`, which is somewhat redundant with the checks here)